project/OsgiCheck.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.io.File
+import sbt.MessageOnlyException
+import scala.collection.mutable.Map
+
+object OsgiCheck {
+  val prog = "osgiCheck"
+
+  // Directories in the repository root that are libraries which should be checked
+  val libDirMatchers = Seq("""^.*(daffodil-\S+)$""".r)
```

Comment:
I'm assuming that if the task is added to each subproject that the task could be run on a single library. The script needs to run over the entire source at once since it is looking for packages that are owned by *other* libraries. If you run osgiCheck against a single library it will always succeed because the owner map won't contain any packages from other libraries. So, does it make sense here to make it a task added at the subproject level?

Reply:
I was viewing a stale pageview of this PR and just saw your implementation below.